core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -112,6 +122,19 @@ public ThisT scanManifestsWith(ExecutorService executorService) {
     return self();
   }
 
+  protected SnapshotMetrics snapshotMetrics() {
```

Comment:
I wonder if this should be `commitMetrics` instead of being specific to a snapshot producer. We certainly care the most about commits that write snapshots, but we may want to at least time and track attempts for other commits as well. This could be done later, but we probably want to get the naming right right away.

What do you think?

Reply:
I'm fine either way. I initially picked `SnapshotMetrics` over `CommitMetrics` because I thought it would cause potential confusion with the commit metrics in Flink, but I think it's less of an issue than I initially thought. That being said, I've renamed everything to `CommitMetrics` / `CommitReport`